### PR TITLE
new packages: dev-{libs,python}/tlsh

### PR DESCRIPTION
**Pull request by laumann**
The dev-util/diffoscope package is capable of comparing contents of archives where files paths are fuzzily matched. This can be useful for comparing distfiles for different versions of the same package.

### CODE REVIEW

**Comment by laumann** [comment]
New version of #25756

File: dev-libs/tlsh/files/tlsh-4.8.2-respect-flags.patch [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+     endif()
+     if(MSVC)
+         set(CMAKE_CXX_FLAGS "/O2")        ## Optimize
+@@ -142,8 +142,7 @@ endif()
+ 
+ # user can override CXX; make sure tests link and load properly regardless of LD_LIBRARY_PATH
+ if(CMAKE_COMPILER_IS_GNUCXX)
+-    set(CMAKE_EXE_LINKER_FLAGS "-static-libgcc")
+-    set(CMAKE_EXE_LINKER_FLAGS "-static-libstdc++")
++    set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -static-libstdc++")
```
Comment (thesamesam) [inline dev-libs/tlsh/files/tlsh-4.8.2-respect-flags.patch:18]: FWIW, this seems to assume GCC will never support libc++ (from LLVM). But not a big deal. I'd leave it as-is.

File: dev-python/tlsh/files/tlsh-4.8.2-setup-sources.patch [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+-else:
+-  tlsh_module = Extension('tlsh', \
+-    sources = ['tlshmodule.cpp', \
+-      join(realpath('..'), 'src', 'tlsh.cpp'), \
+-      join(realpath('..'), 'src', 'tlsh_impl.cpp'), \
+-      join(realpath('..'), 'src', 'tlsh_util.cpp') \
+-    ], \
+-    include_dirs = [join(realpath('..'), 'include')],
+-  )
++tlsh_module = Extension('tlsh', sources=['tlshmodule.cpp'], libraries=['tlsh'])
```
Comment (thesamesam) [inline dev-python/tlsh/files/tlsh-4.8.2-setup-sources.patch:27]: I wonder what the sources= thing is for, if we're using the system copy. Or do I misunderstand the point of the patch?

(Is it just to fix the layout?)

Comment (laumann) [reply dev-python/tlsh/files/tlsh-4.8.2-setup-sources.patch:27]: My understanding is that this is code that creates the python module that can call the shared lib. The module is originally built to include the entire tlsh code in it, but the tlshmodule.cpp only sets up the bindings from python to c++.

File: dev-libs/tlsh/files/tlsh-4.8.2-respect-flags.patch (hunk 1 above)
Comment (thesamesam) [reply dev-libs/tlsh/files/tlsh-4.8.2-respect-flags.patch:18]: Yep!

File: dev-python/tlsh/files/tlsh-4.8.2-setup-sources.patch (hunk 2 above)
Comment (thesamesam) [reply dev-python/tlsh/files/tlsh-4.8.2-setup-sources.patch:27]: Oh okay.

**Comment by thesamesam** [comment]
Ready?

**Comment by laumann** [comment]
> Ready?

Yes! I don't have any outstanding changes to push.

**Comment by laumann** [comment]
Well, I think tests are failing for the dev-python/tlsh package... I should maybe try to fix that.

**Comment by laumann** [comment]
@thesamesam _Now_ I think it's ready - I overrode the `python_test()` function to use the provided test.py runner script (with itself as input). The idea is basically that if it succeeds, then the module works.

**Pull request closed without merging**